### ISSUE TITLE
Cannot figure out how to collect nlog ETW events

### ISSUE THREAD

**Issue by brianhy** [NLog/NLog.Etw#17]
Please see https://stackoverflow.com/questions/50337679/nlog-etw-extension-not-emitting-etw

In short, following the directions on the README, it's unclear how I'm supposed to collect the ETW events being emitted by NLOG.  I was expecting to see the provider id I would reference in a WPRP, but I don't see one, and when I tried setting on, it also didn't work.

**Comment by 304NotModified** [comment]
maybe this could be helpful?

https://github.com/NLog/NLog.Etw/blob/master/NLog.Etw.Tests/EtwSimpleTargetTest.cs

**Comment by snakefoot** [comment]
Seems others have recognized that this NLog.Etw has grown stale. Maybe time for a refresh:

http://zbrad.github.io/tools/NLogEtw/

https://www.nuget.org/packages/ZBrad.NLogEtw/1.0.1

https://github.com/zbrad/nlogetw

**Comment by snakefoot** [comment]
Guess one could make use of the "4th" parameter in `WriteEvent` and then the EventSource-Name becomes dynamic at runtime:

https://stackoverflow.com/questions/43232306/etw-eventsource-in-a-common-component/43375037#43375037

Just need a layout-renderer for fast mapping of EventID. Maybe have a more dynamic mapping. So one can pre-register eventsids:

```
<target type="..." logeventid="${level}" defaultEventId="1">
    <eventid id="2" layout="Warn" />
    <eventid id="3" layout="Error" />
    <eventid id="4" layout="Fatal" />
</target>
```

Maybe a custom-delegate must be created for each custom eventid, or else the EventSource might complain:

https://blogs.msdn.microsoft.com/vancem/2015/10/02/dynamically-defined-events-in-eventsource-v4-6/

Maybe also support `EventOpcode`-enumeration (Custom values must be in the range from 11 through 239):

https://msdn.microsoft.com/en-us/library/system.diagnostics.tracing.eventopcode(v=vs.110).aspx

Maybe also support `EventTask`-enumeration:

https://msdn.microsoft.com/en-us/library/system.diagnostics.tracing.eventtask(v=vs.110).aspx

Maybe also support `Keywords`:

https://stackoverflow.com/questions/30059189/semantic-logging-using-etw-with-custom-keywords-not-logging

**Comment by snakefoot** [comment]
Maybe just add support for providing an EventSource-object from the outside, then all the fancy ETW stuff can be configured by the application before initializing the NLog-target.

Then one would not be restricted by the simple WriteEvent-methods of NLog.

**Comment by 304NotModified** [comment]
Closing this due to inactivity. Please let us know if this still an issue and please provide the requested info -
 thanks!

**Comment by snakefoot** [comment]
Resolved with #19 (NLog.Etw v4.2)
